tests/Predis/Replication/ReplicationStrategyTest.php

```diff
@@ -320,7 +321,7 @@ public function testSetLuaScriptAsReadOperationWorksWithScriptCommand()
         $this->assertFalse($strategy->isReadOperation($command));
 
         $command->setArguments(array(true));
-        $this->assertTrue($strategy->isReadOperation($command));
+        $this->assertFalse($strategy->isReadOperation($command));
```

Comment:
Why?

Reply:
Great, I missed such bogus changes and got them merged in my local branch. :man_facepalming:

A few lines above this change, there's a test-related bug that slipped in while converting tests:

```diff
-        $command = $this->getMock('Predis\Command\ScriptCommand', array('getScript'));
+        $command = $this->getMockBuilder('Predis\Command\ScriptCommand', array('getScript'))->getMock();
```

`\PHPUnit\Framework\TestCase::getMockBuilder()` accepts only one argument (the original class / interface name), but no error is raised when passing a second one. `Predis\Command\ScriptCommand::getScript()` is an abstract method in an abstract class and, at least with PHPUnit 4.x, such methods requiring a mock had to be passed in the second argument of `\PHPUnit_Framework_TestCase::getMock()`.

I noticed this when I was converting some tests from `stale-master` that didn't exist in `v1.1` (so they were not touched by this PR) and I just used `\PHPUnit\Framework\MockObject\MockBuilder::setMethods()` (which is deprecated, but it'll do for now) to have them working again as expected. In this specific case it would be something like this.

```diff
-        $command = $this->getMock('Predis\Command\ScriptCommand', array('getScript'));
+        $command = $this->getMockBuilder('Predis\Command\ScriptCommand')
+            ->setMethods(array('getScript'))
+            ->getMock();
```

**This is an absolute rule**: when not sure why a test fails **just let it fail**! Someone else will fix it. In this case the failure was due to a bogus conversion from old to new PHPUnit, but anyway changing assertions just for the sake of a green build makes a test suite pointless and

Reply:
The issue is the test was always returning `false`. Same as i mentioned below.

Reply:
As explained, the reason why you were getting `FALSE` instead of `TRUE` was due to a bogus change migrating into `getMockBuilder()`, which made the test behave incorrectly.